IPIP/0379-delegated-ipns-http-api.md

```diff
@@ -0,0 +1,82 @@
+- Use of extra headers in `OPTIONS` response to communicate the supported capabilities and limitations, namely:
+  - `X-Ipns-Allow-Max-Size` -- to signal maximum supported IPNS record size
+  - `X-Ipns-Allow-Protobuf` -- to signal whether the server supports ProtocolBuffer formatted records.
+- Streaming interaction is not supported.
+
+### User benefit
+
+The ability of offload naming onto another process or server via an idiomatic and simple HTTP API opens up an opportunity to not only reduce the burden of naming system management, but also introduce alternative naming systems across other routing systems, namely IPNI.
+
+### Compatibility
+
+#### Backwards Compatibility
+
+##### Serialization Format
+
+IPNS records use [ProtocolBuffer serialisation format](../ipns/IPNS.md#record-serialization-format).
+This format is widely in use in IPNS over PubSub and DHT routing systems.
+One of the motivations of this document is to introduce simple to use HTTP APIs and ultimately reduce barrier for interaction across alternative systems.
+Further, interoperability across the existing and HTTP APIs is also desirable in order to reduce the barrier for adoption of the delegated HTTP APIs.
+
+The specification here maintains backwards compatibility in terms of record serialisation, with preference for human-readable formats such as JSON.
+To maximize interoperability with existing ecosystem, the canonical [IPNS record serialization format](../ipns/IPNS.md#record-serialization-format) (`0x0300`) can be requested with content type `application/vnd.ipfs.ipns-record`.
+
+##### Reframe
+
+See [IPIP-337/Backwards Compatibility](0337-delegated-routing-http-api.md#backwards-compatibility).
+
+#### Forwards Compatibility
+
+See [IPIP-337/Forwards Compatibility](0337-delegated-routing-http-api.md#forwards-compatibility).
+
+### Security
+
+All interaction over the APIs should use TLS to protect against third-party observation and tampering.
+Additionally, the IPNS records are signed by the publisher's identity and contain sequence number to avoid replay attacks.
+
+To avoid Denial of Service attack, maximum IPNS record size of `10 KiB` applies.
+Implements are permitted to set a lower limit. If lower than the default maximum, the limit should be discoverable via `OPTIONS` request with header key `X-Ipns-Allow-Max-Size` with value specified as the number of bytes.
+
+Similarly, a client may check if a server supports ProtocolBuffer formatted records by checking the `X-Ipns-Allow-Protobuf` header key in response to `OPTIONS` request. If present the header value must be either `true` or `false` the absence of the header indicates that ProtocolBuffer formatted records are not supported.
+
+Privacy in delegated IPNS is out of scope for this work.
+- The usual JSON parsing rules apply. To prevent potential Denial of Service (DoS) attack, clients should ignore responses larger than 100 providers and introduce a byte size limit that is applicable to their use case.
```

Comment:
I think this is copy-paste from older IPIP  – removing.
```suggestion
- The usual JSON parsing rules apply. To prevent potential Denial of Service (DoS) attack, clients should ignore responses larger than 100 providers and introduce a byte size limit that is applicable to their use case.
```